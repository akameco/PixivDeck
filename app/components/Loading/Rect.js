// @flow
import styled, { keyframes } from 'styled-components'

const loading = keyframes`
	0%,
	40%,
	100% {
		transform: scaleY(0.4);
	}

	20% {
		transform: scaleY(1);
	}
`

const Rect = styled.div`
	background-color: #333;
	height: 100%;
	width: 6px;
	display: inline-block;
	margin: 1px;
	animation: ${loading} 1.2s infinite ease-in-out;
	animation-delay: ${props => props.delay}s;
`

export default Rect
