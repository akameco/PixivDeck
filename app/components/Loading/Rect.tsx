import styled, { keyframes } from 'styled-components'
import { key } from 'styles/styleTheme'

const loading = keyframes`
  0%,
  40%,
  100% {
    transform: scaleY(0.4);
  }

  20% {
    transform: scaleY(1);
  }
`
const Rect = styled.div`
  background-color: ${key('gray')};
  height: 100%;
  width: 6px;
  display: inline-block;
  margin: 1px;
  animation: ${loading} 1.2s infinite ease-in-out;
  animation-delay: ${props => props.delay}s;
`
export default Rect
