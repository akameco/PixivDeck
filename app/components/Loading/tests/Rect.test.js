import React from 'react'
import {shallow, mount} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import Rect from '../Rect'

test('render a prop', () => {
	const delay = -0.1
	const c = mount(<Rect delay={-0.1}/>)
	expect(c.prop('delay')).toEqual(delay)
})

test('snapshot', () => {
	const c = shallow(<Rect/>)
	expect(shallowToJson(c)).toMatchSnapshot()
})
