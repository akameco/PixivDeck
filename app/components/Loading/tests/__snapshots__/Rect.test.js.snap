exports[`test snapshot 1`] = `
<div
  className="ySKiV" />
`;
